unusedtest/provider.go

```diff
@@ -13,21 +13,24 @@ var _ unused.Provider = &Provider{}
 // Provider implements [unused.Provider] for testing purposes.
 type Provider struct {
 	name  string
+	id    string
```

Comment:
I'm not sure we need this field for this test struct, we could just make it always return a hardcoded string. Thoughts?

Reply:
I suppose we don't need it, we could just use a hardcoded string.  Does it make it easier to read/write in the future to have the hardcoded string return my-id?  I'm a little more in favor of it modeling the actual object, but I guess it doesn't really matter